service/matching/matching_engine_test.go

```diff
@@ -3547,6 +3550,93 @@ func (s *matchingEngineSuite) TestPollWorkflowTaskQueueWithRateLimiterError() {
 	s.ErrorIs(err, rateLimiterErr)
 }
 
+func (s *matchingEngineSuite) TestDispatchNexusTask_ValidateTimeoutBuffer() {
+	const ctxTimeout = 2 * time.Second
+	type testCase struct {
+		name      string
+		sleepTime time.Duration
+		assertion func(s *matchingEngineSuite, response *matchingservice.DispatchNexusTaskResponse, err error)
+	}
+
+	testCases := []testCase{
+		{
+			name:      "deadline_exceeded",
+			sleepTime: ctxTimeout - cnexus.NexusTimeoutBuffer,
+			assertion: func(s *matchingEngineSuite, response *matchingservice.DispatchNexusTaskResponse, err error) {
+				s.NoError(err)
```

Comment:
You don't want to use `s.T()` for assertions, you need to use the `T` instance from the subtest to properly attribute the failure to the subtest.
This is a common pitfall when using the testify suites.